tiny-event-sourcing-kafka/src/main/kotlin/ru/quipy/kafka/streams/TopicSubscriptionsManager.kt

```diff
@@ -0,0 +1,134 @@
+
+/**
+ * Creates [KafkaProducerSubscriber] and [KafkaConsumerSubscriber], holds them and allows to destroy them all.
+ * Using this class is a preferable way to create and initialize subscribers.
+ */
+class TopicSubscriptionsManager(
+    private val aggregateEventsStreamManager: AggregateEventStreamManager,
+    private val topicEventsStreamManager: TopicEventStreamManager,
+    private val aggregateRegistry: AggregateRegistry,
+    private val topicRegistry: TopicRegistry,
+    private val eventMapper: EventMapper,
+    private val externalEventMapper: ExternalEventMapper,
+    private val kafkaProperties: KafkaProperties,
+    private val ongoingGroupStorage: OngoingGroupStorage,
+    private val groupRegistry: DomainGroupRegistry,
+    private val externalEventMapperRegistry: ExternalEventMapperRegistry,
+) {
+    private val logger = LoggerFactory.getLogger(TopicSubscriptionsManager::class.java)
+
+    private val subscribers: MutableList<StoppableAndDestructible> = mutableListOf()
+
+    fun <A : Aggregate, T : Topic> createKafkaProducerSubscriber(
+        aggregateClass: KClass<A>,
+        topicEntityClass: KClass<T>,
+        subscriberName: String,
+        retryConf: RetryConf = RetryConf(3, RetryFailedStrategy.SKIP_EVENT)
+    ): KafkaProducerSubscriber<A, T> {
+        logger.info("Start creating subscription to aggregate ${aggregateClass.simpleName}, on topic ${topicEntityClass.simpleName}, subscriber name $subscriberName")
+
+        val internalEventInfo = aggregateRegistry.getEventInfo(aggregateClass)
+            ?: throw IllegalArgumentException("Couldn't find aggregate class ${aggregateClass.simpleName} in registry")
+
+        val externalEventInfo = topicRegistry.getExternalEventInfo(topicEntityClass)
+            ?: throw IllegalArgumentException("Couldn't find topic class ${topicEntityClass.simpleName} in registry")
+
+        val stream = aggregateEventsStreamManager.createEventStream(
+            "$subscriberName-stream",
+            aggregateClass,
+            retryConf
+        )
+
+        val topicName = topicRegistry.basicTopicInfo(topicEntityClass)?.topicName.toString()
+
+        val kafkaTopicCreator = KafkaTopicCreator()
+
+        val topicConfig = TopicConfig(
+            kafkaProperties.bootstrapServers!!,
+            topicName,
+            kafkaProperties.partitions,
+            kafkaProperties.replicationFactor
+        )
+
+        kafkaTopicCreator.createTopicIfNotExists(topicConfig)
+
+        val kafkaProducer = KafkaEventProducer<T>(topicName, kafkaProperties)
+
+        val ongoingGroupManager = OngoingGroupManager(groupRegistry, ongoingGroupStorage, externalEventMapperRegistry, eventMapper, externalEventMapper)
+
+        val subscriber = KafkaProducerSubscriber(
+            stream,
+            kafkaProducer,
+            ongoingGroupManager,
+            eventMapper,
+            internalEventInfo::getEventTypeByName
+        )
+
+        if (subscribers.contains(subscriber)) {
```

Comment:
nit: `subscribers` can be a Set
Also, is there no way to check existence of the subscriber before the creation of a new one?

Reply:
Don't see such a way